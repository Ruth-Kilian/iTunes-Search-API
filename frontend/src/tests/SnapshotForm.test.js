/* Snapshot test -> to check if the <Form> component renders correctly when it is passed the setSearchResults prop */

import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Form from "../components/SearchForm";

describe("Search", () => {
  test("renders correctly", () => {
    // mock function to be passed as a prop -> simulates the behaviour
    const setSearchResults = jest.fn();
    // renderer.create() used to create an instance of the component
    const component = renderer.create(
      // Form is wrapped in a BrowserRouter so that useHistory hook can be used
      <BrowserRouter>
        <Form setSearchResults={setSearchResults} />
      </BrowserRouter>
    );
    // compares the rendered component to a previously stored snapshot
    expect(component).toMatchSnapshot();
  });
});
