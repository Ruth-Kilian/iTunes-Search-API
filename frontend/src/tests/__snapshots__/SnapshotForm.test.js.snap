// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div>
  <h1>
    iTunes API
  </h1>
  <p>
    <a
      href="/favourites"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
    >
      Favourites List
    </a>
  </p>
  <div
    className="form-container"
  >
    <div
      className="form-row"
    >
      <label>
        Search Term:
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-row"
    >
      <label>
        Number of Items:
      </label>
      <input
        onChange={[Function]}
        type="number"
        value={10}
      />
    </div>
    <div
      className="form-row"
    >
      <label>
        Media Type:
      </label>
      <select
        onChange={[Function]}
        value="all"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="song"
        >
          Song
        </option>
        <option
          value="audiobook"
        >
          Audiobook
        </option>
        <option
          value="shortFilm"
        >
          Short Film
        </option>
        <option
          value="tvShow"
        >
          tvShow
        </option>
        <option
          value="software"
        >
          Software
        </option>
        <option
          value="ebook"
        >
          Ebook
        </option>
      </select>
    </div>
    <div
      className="form-row"
    >
      <button
        className="buttonSearch"
        onClick={[Function]}
      >
        SearchForm
      </button>
    </div>
  </div>
</div>
`;
